elastic/kibana
# [Fleet] Refactor setup to load default packages/policies with preconfiguration

## Summary

Closes #97016 

This removes the bespoke functions for installing default packages and configuring the default agent policies, and instead sets these as the default Kibana Config values for the preconfiguration API.

### Testing

### Default setup

1. Remove **all references** to `xpack.fleet` from your `kibana.dev.yml`
2. Start Kibana and navigate to the Fleet app
3. Ensure that the Default Agent Policy and Default Fleet Server Agent Policy have been created
4. Ensure that the System, Endpoint Security, Elastic Agent, and Fleet Server integrations are installed

### Overriding the defaults

1. Restart your `yarn es snapshot` to clear your cluster, then set your `kibana.dev.yml` to:

```yaml
xpack.fleet.packages:
  - name: system
    version: latest
  - name: endpoint
    version: latest
  - name: elastic_agent
    version: latest
  - name: fleet_server
    version: 0.2.0
  - name: apache
    version: latest
```

Ensure that all five of these integrations are installed, and that Fleet Server is installed with an outdated version.

2. Restart `yarn es snapshot` again, then set your `kibana.dev.yml` to:

```yaml
xpack.fleet.agentPolicies:
  - name: Alternative Default Agent Policy
    id: 1
    namespace: test
    is_default: true
    monitoring_enabled: []
    package_policies:
      - package:
          name: system
        name: System Integration
  - name: Alternative Default Fleet Server Agent Policy
    id: 2
    namespace: test
    is_default_fleet_server: true
    package_policies:
      - package:
          name: fleet_server
        name: Fleet Server Integration
```

Ensure that the default policies are created with these alternative names and that agent monitoring is disabled for the default policy, but enabled for the default fleet server policy. Try to delete them, and ensure that they **fail to be deleted**.

3. Restart `yarn es snapshot` again, then change your `kibana.dev.yml` to:

```yaml
xpack.fleet.agentPolicies:
  - name: Agent Policy With No Default
    id: 1
    namespace: test
    package_policies:
      - package:
          name: system
        name: System Integration
```

Ensure that Fleet starts with only this one policy, and that it can be deleted.


### Checklist
